tests/web/test_transactions.py

```diff
@@ -19,6 +21,59 @@ def test_get_transaction_endpoint(b, client, user_pk):
 
 
 @pytest.mark.bdb
+def test_get_backlog_tx(client, backlog_tx):
+    from bigchaindb import Bigchain
+    from bigchaindb.web.views.transactions import TransactionApi
+    path = url_for(TransactionApi.__name__.lower(), tx_id=backlog_tx.id)
+    res = client.get(path=path,
+                     query_string={'status': Bigchain.TX_IN_BACKLOG})
+    assert res.status_code == 200
+    assert res.json == backlog_tx.to_dict()
+
+
+@pytest.mark.bdb
+def test_get_undecided_tx(client, undecided_tx):
+    from bigchaindb import Bigchain
+    from bigchaindb.web.views.transactions import TransactionApi
+    path = url_for(TransactionApi.__name__.lower(), tx_id=undecided_tx.id)
+    res = client.get(path=path,
+                     query_string={'status': Bigchain.BLOCK_UNDECIDED})
+    assert res.status_code == 200
+    assert res.json == undecided_tx.to_dict()
+
+
+@pytest.mark.bdb
+def test_get_valid_tx(client, valid_block):
+    from bigchaindb import Bigchain
+    from bigchaindb.web.views.transactions import TransactionApi
+    tx = valid_block.transactions[0]
+    path = url_for(TransactionApi.__name__.lower(), tx_id=tx.id)
+    res = client.get(path=path, query_string={'status': Bigchain.BLOCK_VALID})
+    assert res.status_code == 200
+    assert res.json == tx.to_dict()
+
+
+@pytest.mark.bdb
+@pytest.mark.parametrize('tx_status', ('valid', 'backlog', 'undecided'))
+def test_get_with_status_param_not_found(client, tx_status, caplog):
+    from bigchaindb.web.views.transactions import TransactionApi
+    path = url_for(TransactionApi.__name__.lower(), tx_id=1)
+    res = client.get(path=path, query_string={'status': tx_status})
+    expected_error_message = (
+        'Transaction with id: "{}" and '
+        'status: "{}" could not be found.'
+    ).format(1, tx_status)
+    expected_status_code = 404
+    assert res.status_code == expected_status_code
+    assert len(res.json) == 2
+    assert res.json['status'] == expected_status_code
+    assert res.json['message'] == expected_error_message
+    assert len(caplog.records[-1].args)
+    assert caplog.records[-1].args['status'] == expected_status_code
+    assert caplog.records[-1].args['message'] == expected_error_message
```

Comment:
We *could* also have a test that adds some transactions / blocks and tests that queries for empty statuses still return empty.

As far as I can tell, this test is only checking that we get nothing back for an empty database.